src/js/message.js

```diff
@@ -0,0 +1,138 @@
+import construct from './construct-element';
+import { throwError } from './helpers';
+
+class Message {
+	/**
+ * Class constructor.
+ * @param {HTMLElement} [messageElement] - The message element in the DOM
+ * @param {Object} [options={}] - An options object for configuring the message
+ */
+	constructor(messageElement, options) {
+		this.messageElement = messageElement;
+
+		//Default options
+		const messageClass = options && options.messageClass ? options.messageClass : 'o-message';
+		const type = options && options.type ? options.type : 'alert';
+		const status = options && options.status ? options.status : null;
+
+		this.opts = Object.assign({}, {
+			autoOpen: true,
+			messageClass,
+			type,
+			typeClass: `${messageClass}--${type}`,
+			status,
+			statusClass: options && options.status ? `${messageClass}--${options.status}` : null,
+			parentElement: null,
+			content: {
+				highlight: null,
+				detail: '&hellip;',
+				additionalInfo: false
+			},
+			actions: {
+				primary: {
+					text: null,
+					url: '#'
+				},
+				secondary: {
+					text: null,
+					url: '#'
+				}
+			},
+			close: true
+		}, options);
+
+		this.render();
+
+		if (this.opts.autoOpen) {
+			this.open();
+		} else {
+			this.close();
+		}
+	}
+
+	/**
+	 * Render the message.
+	 */
+	render () {
+		// If the message element is not an HTML Element, or if a parent element has been specified, build a new message element
+		if (!(this.messageElement instanceof HTMLElement) || this.opts.parentElement) {
+			this.messageElement = this.constructMessageElement();
+
+			// attach oMessage to specified parentElement or default to document body
+			let element = this.opts.parentElement ? document.querySelector(this.opts.parentElement) : document.body;
+			element.appendChild(this.messageElement);
+		}
+
+		// if the message is 'alert-inner' it shouldn't be close-able
+		if (this.messageElement.matches("[class*='-inner']")) {
+			this.opts.close = false;
+		};
+
+		if (this.opts.close) {
+			let closeButton = construct.closeButton(this.opts);
+
+			// Add event listeners
+			closeButton.addEventListener('click', event => {
+				event.preventDefault();
+				this.close();
+			});
+
+			this.messageElement.lastElementChild.appendChild(closeButton);
```

Comment:
Would this be better in `constructMessageElement`?

Reply:
It wouldn't be called if the message was set up declaratively (I've removed all of the close buttons from the markup)